src/python/arcor2_arserver/rpc/project.py

```diff
@@ -68,126 +67,125 @@ async def managed_project(project_id: str, make_copy: bool = False) -> AsyncGene
 @project_needed
 async def cancel_action_cb(req: srpc.p.CancelAction.Request, ui: WsClient) -> None:
 
-    assert glob.LOCK.scene
-    assert glob.LOCK.project
-
-    if not glob.RUNNING_ACTION:
-        raise Arcor2Exception("No action is running.")
+    async with ctx_write_lock(glob.LOCK.SpecialValues.RUNNING_ACTION, glob.USERS.user_name(ui)):
```

Comment:
A `RUNNING_ACTION` je potřeba proč? Nestačilo by zamknout objekt, ke kterému ta akce patří (jehož metoda se volá)?

Reply:
Ano, v případě rušení akce to je možné, upravím

Reply:
Jde mi o to, abychom se co nejvíc vyhnuli nějakým speciálním hodnotám...

Reply:
> Taky si nejsem jistý, jestli je tohle správné řešení v návaznosti dnešní zprávy o možnosti více kalibrací, protože zamykání téhle proměnné umožní jen jedinou kalibraci na serveru?

To že poběží víc instancí kalibrační služby by nemělo nijak ovlivnit co se děje na serveru. V zásadě se můžou kalibrovat tři věci:
- Robot (měl by být zamčený objekt robota + kamery kterou se to provádí).
- Kamera (zamčený objekt kamery).
- Kalibruje se pozice UI (tabletu) - tady se nic nezamyká.